modules/core/src/main/java/org/apache/ignite/internal/util/GridLogThrottle.java

```
@@ -231,6 +261,26 @@ private static boolean replace(IgniteBiTuple<Class<? extends Throwable>, String>
         return errors.replace(t, oldStamp, newStamp);
     }
 
+    /**
+     * Method iterates though man and removes outdated entries (compares entry timestamp with current
+     * timestamp minus timeout). It protects the map from causing memory leak.
+     */
+    private static void cleanUpOldEntries() {
+        long curTs = U.currentTimeMillis();
+
+        Iterator<Map.Entry<IgniteBiTuple<Class<? extends Throwable>, String>, Long>> itr = errors.entrySet().iterator();
```

Is it fail fast or fail safe iterator?